warehouse/query-core/src/main/java/datawave/query/transformer/ExcerptTransform.java

```diff
@@ -314,29 +314,62 @@ private Set<Excerpt> getExcerpts(PhraseIndexes phraseIndexes) {
      * @return the excerpt
      */
     private String getExcerpt(String field, int start, int end, Range range, ArrayList<String> hitTermValues) {
+        int prevLeftWordsSkipped = 0;
+        int currLeftWordsSkipped = 0;
+        boolean leftLock = false;
+        int prevRightWordsSkipped = 0;
+        int currRightWordsSkipped = 0;
+        boolean rightLock = false;
+        int timesToTry = 100;
         excerptIteratorOptions.put(TermFrequencyExcerptIterator.FIELD_NAME, field);
-        excerptIteratorOptions.put(TermFrequencyExcerptIterator.START_OFFSET, String.valueOf(start));
-        excerptIteratorOptions.put(TermFrequencyExcerptIterator.END_OFFSET, String.valueOf(end));
-        try {
-            excerptIterator.init(source, excerptIteratorOptions, env);
-            excerptIterator.seek(range, Collections.emptyList(), false);
-            if (excerptIterator.hasTop()) {
-                Key topKey = excerptIterator.getTopKey();
-                String[] parts = topKey.getColumnQualifier().toString().split(Constants.NULL);
-                // The column qualifier is expected to be field\0phrase.
-                if (parts.length == 2) {
-                    return getHitPhrase(hitTermValues, parts);
+        for (int i = 0; i <= timesToTry; i++) {
```

Comment:
I really do not like this multiple pass thing.  The number of terms in a document can be HUGE.  So making this pass multiple times will cause us significant performance issues.  How about filling the initial array of terms to be significantly outside the bounds of the initial request, and then trim down the excerpt to what we need.  This would be significantly less expensive.

Reply:
I've done a hybrid of the ideas.